From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nomorecoolnicknames <nomorecoolnicknames@users.noreply.github.com>
Date: Tue, 28 Jul 2026 13:42:50 +0300
Subject: [PATCH 2/5] symbols: MTK_SYMBOLS_GUI_ONLY builds only the
 libgui/libui shims

The audio and camera shims link libraries (libdpframework and others)
that a device may not have. A device tree that includes this makefile
with MTK_SYMBOLS_GUI_ONLY=true gets libmtkshim_gui and libmtkshim_ui
only.
---
 symbols/Android.mk | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/symbols/Android.mk b/symbols/Android.mk
index 5cb4c12..5507069 100644
--- a/symbols/Android.mk
+++ b/symbols/Android.mk
@@ -13,6 +13,17 @@ LOCAL_PROPRIETARY_MODULE := true
 LOCAL_MODULE_TAGS := optional
 include $(BUILD_SHARED_LIBRARY)
 
+include $(CLEAR_VARS)
+LOCAL_SRC_FILES := ui.cpp
+LOCAL_SHARED_LIBRARIES := libbinder libui libutils
+LOCAL_MODULE := libmtkshim_ui
+LOCAL_MULTILIB := both
+LOCAL_PROPRIETARY_MODULE := true
+include $(BUILD_SHARED_LIBRARY)
+
+# The audio and camera shims link modules that not every device has.
+ifneq ($(MTK_SYMBOLS_GUI_ONLY),true)
+
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES := audio.cpp
 LOCAL_SHARED_LIBRARIES := libmedia
@@ -31,10 +42,4 @@ LOCAL_MODULE_CLASS := SHARED_LIBRARIES
 LOCAL_PROPRIETARY_MODULE := true
 include $(BUILD_SHARED_LIBRARY)
 
-include $(CLEAR_VARS)
-LOCAL_SRC_FILES := ui.cpp
-LOCAL_SHARED_LIBRARIES := libbinder libui libutils
-LOCAL_MODULE := libmtkshim_ui
-LOCAL_MULTILIB := both
-LOCAL_PROPRIETARY_MODULE := true
-include $(BUILD_SHARED_LIBRARY)
+endif
